src/node/rpc/test/proposal_id_test.cpp

```diff
@@ -120,14 +120,14 @@ class NullTxHistoryWithOverride : public ccf::NullTxHistory
       forced = false;
       return {
         {term_of_last_version, forced_version},
-        crypto::Sha256Hash(std::to_string(version)),
+        crypto::Sha256Hash(CBuffer(std::to_string(version))),
         term_of_next_version};
     }
     else
     {
       return {
         {term_of_last_version, version},
-        crypto::Sha256Hash(std::to_string(version)),
+        crypto::Sha256Hash(CBuffer(std::to_string(version))),
```

Comment:
I think I would have added a const std::string& variant of the ctor rather than use CBuffer, but ymmv,

Reply:
I added that constructor in this PR, but it expects the string to be hex-encoded, which isn't the case here. Using a `version` for the hash is only seen in the context of this test, hence the hack here.